File: call-in-script/arithmetic-functions.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import os
import math
import subprocess

os.putenv("SAC_DISPLAY_COPYRIGHT", "0")

p = subprocess.Popen(['sac'], stdin=subprocess.PIPE)

var1 = (1+2)*3/4
var2 = math.sqrt(9)
var3 = math.sin(30*math.pi/180)

s = ""
s += "fg seis \n"
s += "rmean; rtr; taper \n"
s += "bp c %f %f \n" % (var3, var1)
s += "w seis.SAC \n"
s += "q \n"

p.communicate(s.encode())

File: call-in-script/arithmetic-functions.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Math::Trig;
$ENV{SAC_DISPLAY_COPYRIGHT}=0;

my $var1 = (1+2)*3/4;
open(SAC, "| sac") or die "Error in opening sac\n";
print SAC "fg seis \n";
print SAC "rmean; rtr; taper \n";
my $var2 = sqrt(9);
my $var3 = sin(30*pi/180);
print SAC "bp c $var3 $var1 \n";
print SAC "q \n";
close(SAC);
